cli/azd/pkg/input/console.go

```diff
@@ -63,14 +67,35 @@ func (c *AskerConsole) SetWriter(writer io.Writer) {
 
 // Prints out a message to the underlying console write
 func (c *AskerConsole) Message(ctx context.Context, message string) {
-	// Only write to the console during interactive & non-formatted responses.
-	if c.interactive && (c.formatter == nil || c.formatter.Kind() == output.NoneFormat) {
+	if c.formatter != nil && c.formatter.Kind() == output.JsonFormat {
+		// we do not expect formatter.Format to fail on valid input. Discard the error to appease the linter (note
+		// that we implicitly discard the error from fmt.Fprintln later, as well).
+		_ = c.formatter.Format(c.consoleMessageForMessage(message), c.writer, nil)
```

Comment:
Do we need to ensure the `json` is minified into a single line?

Reply:
I don't think so.  I think any consumer is just going to need to read an object at a time from `stdout` and the intermixed newlines won't cause an issue.  FWIW, in an earlier iteration I did just write this as a single line (eschewing the formatter API and just calling `json.Marshal` myself so there was no indentation) but I realized that would be different from the final result for most of our commands (where we do format the object across multiple lines and indented).

This also makes reading the JSON output as a human so much easier, which I think is a nice plus.